pkg/bundle/reader.go

```diff
@@ -116,7 +116,7 @@ func readTarLayer(l v1.Layer) ([]byte, error) {
 	}
 
 	contents := make([]byte, header.Size)
-	if _, err := treader.Read(contents); err != nil && err != io.EOF {
+	if _, err := io.ReadFull(treader, contents); err != nil && err != io.EOF {
```

Comment:
that's not the same tho, you are going to read the whole content in memeory instead of stream it now....

Reply:
The extracted content would have to fit into an etcd, chunk? So there should be a maximum size of the blob, right?

If we want to prevent this from getting too big, we could potentially add validation into `tkn bundle push` to warn/fail if the size of the extracted content is too big.

Reply:
But yes, there is definitely a follow-up to be done here, but this does fixes some existing issue.

Reply:
https://github.com/tektoncd/cli/issues/2434

Reply:
Might want to have guard on `header.Size` not being over 1.5MB (https://etcd.io/docs/v3.5/dev-guide/limit/), might introduce a denial of service via crafted compressed layer (zip bomb)